src/EventSubscriber/OgEventSubscriber.php

```diff
@@ -59,18 +78,90 @@ public static function getSubscribedEvents() {
    */
   public function provideDefaultOgPermissions(PermissionEventInterface $event) {
     $event->setPermissions([
-      'update group' => [
+      new GroupPermission([
+        'name' => 'update group',
         'title' => t('Edit group'),
         'description' => t('Edit the group. Note: This permission controls only node entity type groups.'),
         'default roles' => [OgRoleInterface::ADMINISTRATOR],
-      ],
-      'administer group' => [
+      ]),
+      new GroupPermission([
+        'name' => 'administer group',
         'title' => t('Administer group'),
         'description' => t('Manage group members and content in the group.'),
         'default roles' => [OgRoleInterface::ADMINISTRATOR],
         'restrict access' => TRUE,
-      ],
-    ] + $this->permissionManager->getPermissionList($event->getEntityTypeId(), $event->getBundleId()));
+      ]),
+    ]);
+
+    // Add a list of generic CRUD permissions for all group content.
+    $group_content_permissions = $this->getDefaultEntityOperationPermissions($event->getGroupContentBundleIds());
+    $event->setPermissions($group_content_permissions);
+  }
+
+  /**
+   * Provides default permissions for the Node entity.
+   *
+   * @param \Drupal\og\Event\PermissionEventInterface $event
+   *   The OG permission event.
+   */
+  public function provideDefaultNodePermissions(PermissionEventInterface $event) {
+    $bundle_ids = $event->getGroupContentBundleIds();
+
+    if (!array_key_exists('node', $bundle_ids)) {
+      return;
+    }
+
+    $permissions = [];
+    $bundle_info = $this->entityTypeBundleInfo->getBundleInfo('node');
+
+    foreach ($bundle_ids['node'] as $bundle_id) {
+      $args = ['%type_name' => $bundle_info[$bundle_id]['label']];
+      $permission_values = [
+        [
+          'name' => "create $bundle_id content",
+          'title' => $this->t('%type_name: Create new content', $args),
+          'operation' => 'create',
+          'default roles' => [OgRoleInterface::ADMINISTRATOR, OgRoleInterface::AUTHENTICATED],
```

Comment:
Question: should normal members by default get the permission to create new group content, and edit and delete their own content? Or is this something that is only reserved for administrators by default?